### PR TITLE
Add support for hover pointer events

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

**Comment by abarth** [comment]
Travis will be red because this needs an engine update as well.

File: packages/flutter/lib/src/gestures/events.dart [hunk 1]
```diff
@@ -311,6 +316,60 @@ class PointerRemovedEvent extends PointerEvent {
...
+  ///
+  /// All of the argument must be non-null.
+  const PointerHoverEvent({
+    Duration timeStamp: Duration.ZERO,
+    PointerDeviceKind kind: PointerDeviceKind.touch,
+    int device: 0,
+    Point position: Point.origin,
+    Offset delta: Offset.zero,
+    int buttons: 0,
+    bool down: false,
```
Comment (Hixie) [inline packages/flutter/lib/src/gestures/events.dart]: remove down, hover can't be down

File: packages/flutter/lib/src/gestures/events.dart [hunk 2]
```diff
@@ -311,6 +316,60 @@ class PointerRemovedEvent extends PointerEvent {
...
+    bool obscured: false,
+    double pressure: 1.0,
```
Comment (Hixie) [inline packages/flutter/lib/src/gestures/events.dart]: remove pressure*, hover can't be pressured

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/gestures/events.dart [hunk 3]
```diff
@@ -364,6 +431,7 @@ class PointerMoveEvent extends PointerEvent {
     Duration timeStamp: Duration.ZERO,
     int pointer: 0,
     PointerDeviceKind kind: PointerDeviceKind.touch,
+    int device: 0,
     Point position: Point.origin,
     Offset delta: Offset.zero,
     int buttons: 0,
```
Comment (abarth) [inline packages/flutter/lib/src/gestures/events.dart]: done

**Pull request merged**